shared/src/main/scala/weaponregex/model/mutation/TokenMutator.scala

```diff
@@ -9,7 +9,13 @@ trait TokenMutator {
 
   final def apply(token: RegexTree): Seq[String] = mutate(token)
 
+  /** I'm noticing most of the implementations for this function have a pattern match on the expected tree type of the mutator.
+    * Perhaps the mutate function could be generalized so a mutator only accepts the expected Tree type?
+    * I would also expect this to return a `Seq[RegexTree]`
```

Comment:
Since the muated `RegexTree` won't be used anywhere, returning String intsead of `RegexTree` allows it to skip the actual creation of the mutated of the tree (which requires copying the subtrees around), and directly alter the tree building procress (e.g. via `buildWith`). This can be seen as the token mutators mutate the tree on "build-level" instead of tree-level.

Reply:
The tree is immutable, right? So copying will only change a few objects. That should not give a noticeable performance impact at all. For the API it makes more sense to have a mutated RegexTree return another RegexTree. That it will build into a string somewhere else is unrelated to the TokenMutator